guppylang/tys/parsing.py

```diff
@@ -117,10 +128,10 @@ def _try_parse_defn(node: AstNode, globals: Globals) -> Definition | None:
     match node:
         case ast.Name(id=x):
             if x not in globals:
-                raise GuppyError(VarNotDefinedError(node, x))
+                return None
```

Comment:
Small refactor: If `x` is not a global, we shouldn't error immediately because it could also refer to a generic parameter. This is now checked above